src/spatialdata_plot/pl/render.py

```diff
@@ -416,9 +449,15 @@ def _render_points(
                 y_ext = [min(y_ext[0], previous_ylim[1]), max(y_ext[1], previous_ylim[0])]
             else:  # case for e.g. labels
                 y_ext = [min(y_ext[0], previous_ylim[0]), max(y_ext[1], previous_ylim[1])]
-        # round because we need integers
-        plot_width = int(np.round(x_ext[1] - x_ext[0]))
-        plot_height = int(np.round(y_ext[1] - y_ext[0]))
+
+        # compute canvas size in pixels close to the actual image size to speed up computation
+        plot_width = x_ext[1] - x_ext[0]
+        plot_height = y_ext[1] - y_ext[0]
+        plot_width_px = int(round(fig_params.fig.get_size_inches()[0] * fig_params.fig.dpi))
+        plot_height_px = int(round(fig_params.fig.get_size_inches()[1] * fig_params.fig.dpi))
+        factor = np.min([plot_width / plot_width_px, plot_height / plot_height_px])
+        plot_width = int(np.round(plot_width / factor))
+        plot_height = int(np.round(plot_height / factor))
```

Comment:
I'd consider bundling this code in a private function since it's duplicate from above.